### PR TITLE
[92][95][98][104][107][110] Implement account page and CastingRejectedForm

### PR DESCRIPTION
**Pull request by CDGstudent**
## What did you do

- add PendingUserPage, PendingUserCard, UserConfirmationCard, UserRejectionCard, usePendingUserData, RejectedForm for Casting

## Demo

-

### CODE REVIEW

**Comment by PKhing** [comment]
- padding
image
image
- space between cards is too wide

**Comment by PKhing** [comment]
- border radius again
- reason should be multiple lines (see description field in edit profile page for an example)
image

**Comment by PKhing** [comment]
responsive
image

File: apps/web/src/modules/account/pages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { CircularProgress, Typography } from '@mui/material'
+import withGuard from 'common/hoc/withGuard'
+import React from 'react'
+
+import PendingUserCard from '../component/PendingUserCard/Body'
+import usePendingUserData from '../hooks/usePendingUserData'
+import { CardsContainer } from './styled'
+
+const PendingUserPage = () => {
+  const pendingUserData = usePendingUserData()
```
Comment (PKhing) [inline apps/web/src/modules/account/pages/index.tsx:11]:
```suggestion
  const {pendingUserData ,setModalReason บลา ๆ } = usePendingUserData()
```

File: apps/web/src/modules/account/pages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          <Typography color="#00000061">ไม่พบการขอสมัครเข้าสู่ระบบ</Typography>
+        ) : (
+          pendingUserData.pendingUserData.map((user) => (
+            <PendingUserCard
+              setReason={pendingUserData.setModalReason}
+              setId={pendingUserData.setModalId}
+              accept={pendingUserData.acceptUser}
+              reject={pendingUserData.rejectUser}
+              data={user}
+            />
```
Comment (PKhing) [inline apps/web/src/modules/account/pages/index.tsx:25]: add key here (please read this https://www.geeksforgeeks.org/reactjs-keys/)

File: apps/web/src/modules/rejected/casting/type.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export interface EditCastingInfoFormProps {
+  initialData: any
```
Comment (PKhing) [inline apps/web/src/modules/rejected/casting/type.ts:2]: P'Yu mai took jai sing nee (type any is not acceptable here)

**Review by PKhing — CHANGES_REQUESTED** [review]
Currently the structure is
```
Component
  Body
    index << the main component is exported from this file
  Footer
    index
  Header
    index
```

but it should be
```
Component
  Body
    index
  Footer
    index
  Header
    index
index << the main component is exported from this file
```

File: apps/web/src/modules/account/component/PendingUserCard/Body/index.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            หนังสือรับรองการทำงาน
+          </Typography>
+        )}
+        {props.data.type === 'ACTOR' && (
+          <Typography variant="button" color="primary">
+            รูปถ่ายบัตรประชาชน/พาสปอร์ต
+          </Typography>
+        )}
+      </ResumeDownloadButton>
+      <>
```
Comment (miello) [inline apps/web/src/modules/account/component/PendingUserCard/Body/index.tsx:48]: Fragment is not necessary in this case, please consider remove

**Review by miello — CHANGES_REQUESTED** [review]

**Comment by PKhing** [comment]
components
image

File: apps/web/src/modules/account/component/PendingUserCard/PendingUserCardFooter/index.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import { PendingUserCardFooterProps } from './type'
+
+const PendingUserCardFooter = ({
+  userId,
+  accept,
+  reject,
+  setId,
+  setReason,
+}: PendingUserCardFooterProps) => {
+  const [openAccept, setOpenAccept] = React.useState(false)
```
Comment (PKhing) [inline apps/web/src/modules/account/component/PendingUserCard/PendingUserCardFooter/index.tsx:16]: use useSwitch (read from change log)

File: apps/web/src/modules/account/component/PendingUserCard/PendingUserCardFooter/index.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  const handleOpenAccept = () => {
+    setId(userId)
+    setOpenAccept(true)
+  }
+  const handleCloseAccept = () => setOpenAccept(false)
+
+  const [openReject, setOpenReject] = React.useState(false)
```
Comment (PKhing) [inline apps/web/src/modules/account/component/PendingUserCard/PendingUserCardFooter/index.tsx:23]: same as above

File: apps/web/src/modules/account/component/PendingUserCard/PendingUserCardFooter/index.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  accept,
+  reject,
+  setId,
+  setReason,
+}: PendingUserCardFooterProps) => {
+  const [openAccept, setOpenAccept] = React.useState(false)
+  const handleOpenAccept = () => {
+    setId(userId)
+    setOpenAccept(true)
+  }
```
Comment (PKhing) [inline apps/web/src/modules/account/component/PendingUserCard/PendingUserCardFooter/index.tsx:20]: useCallback (for all function in this page)

File: apps/web/src/modules/account/component/PendingUserCard/PendingUserCardFooter/index.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  return (
+    <FooterContainer>
+      <Button color="error" onClick={handleOpenReject}>
+        <Typography variant="button">ปฏิเสธ</Typography>
+      </Button>
+      <Modal
+        open={openReject}
+        onClose={handleCloseReject}
+        aria-labelledby="modal-modal-title"
+        aria-describedby="modal-modal-description"
```
Comment (PKhing) [inline apps/web/src/modules/account/component/PendingUserCard/PendingUserCardFooter/index.tsx:48]: is this necessary ?

File: apps/web/src/modules/account/component/PendingUserCard/PendingUserCardFooter/index.tsx [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        <UserRejectionModal setReason={setReason} reject={handleReject} />
+      </Modal>
+      <Button color="success" onClick={handleOpenAccept}>
+        <Typography variant="button">อนุมัติ</Typography>
+      </Button>
+      <Modal
+        open={openAccept}
+        onClose={handleCloseAccept}
+        aria-labelledby="modal-modal-title"
+        aria-describedby="modal-modal-description"
```
Comment (PKhing) [inline apps/web/src/modules/account/component/PendingUserCard/PendingUserCardFooter/index.tsx:59]: is this necessary ?

File: apps/web/src/modules/account/component/PendingUserCard/index.tsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
+import { UserType } from '@modela/database'
```
Comment (miello) [inline apps/web/src/modules/account/component/PendingUserCard/index.tsx:1]:
```suggestion
import { UserType } from '@modela/dtos'
```

File: apps/web/src/modules/account/component/PendingUserCard/PendingUserCardHeader/index.tsx [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+            display: 'flex',
+            justifyContent: 'center',
+            marginLeft: '16px',
+            flexDirection: 'column',
+          }}
+        >
+          <Typography variant="h6">
+            {firstName} {middleName} {lastName}
+          </Typography>
+          {props.type === 'CASTING' && (
```
Comment (PKhing) [inline apps/web/src/modules/account/component/PendingUserCard/PendingUserCardHeader/index.tsx:34]: enum

File: apps/web/src/modules/account/component/PendingUserCard/index.tsx [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+      {type === UserType.CASTING && (
+        <Typography variant="subtitle2" color="#00000099">
+          เลขจดทะเบียน: {companyId}
+        </Typography>
+      )}
+      {type === UserType.ACTOR && (
+        <Typography variant="subtitle2" color="#00000099">
+          เลขบัตรประจำตัวประชาชน/เลขพาสปอร์ต: {ssn}
+        </Typography>
+      )}
```
Comment (PKhing) [inline apps/web/src/modules/account/component/PendingUserCard/index.tsx:30]:
```suggestion
      {type === UserType.CASTING ? (
        <Typography variant="subtitle2" color="#00000099">
          เลขจดทะเบียน: {companyId}
        </Typography>
      ) : (
        <Typography variant="subtitle2" color="#00000099">
          เลขบัตรประจำตัวประชาชน/เลขพาสปอร์ต: {ssn}
        </Typography>
      )}
```

File: apps/web/src/modules/account/component/UserConfirmationModal/UserConfirmationModalHeader/index.tsx [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+
+const UserConfirmationModalHeader = () => {
+  return (
+    <HeaderContainer>
+      <Typography variant="h5">ยืนยันการอนุมัติ</Typography>
+    </HeaderContainer>
+  )
+}
+
+export default UserConfirmationModalHeader
```
Comment (PKhing) [inline apps/web/src/modules/account/component/UserConfirmationModal/UserConfirmationModalHeader/index.tsx:14]: เอาไปรวมกับ index

File: apps/web/src/modules/account/component/UserRejectionModal/UserRejectionModalHeader/index.tsx [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+
+const UserRejectionModalHeader = () => {
+  return (
+    <HeaderContainer>
+      <Typography variant="h5">เหตุผลในการปฏิเสธ</Typography>
+    </HeaderContainer>
+  )
+}
+
+export default UserRejectionModalHeader
```
Comment (PKhing) [inline apps/web/src/modules/account/component/UserRejectionModal/UserRejectionModalHeader/index.tsx:14]: รวมกับ index

File: apps/web/src/modules/account/pages/index.tsx [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+import { CardsContainer } from './styled'
+
+const PendingUserPage = () => {
+  const {
+    pendingUserData,
+    rejectUser,
+    acceptUser,
+    setModalId,
+    setModalReason,
+  } = usePendingUserData()
```
Comment (miello) [inline apps/web/src/modules/account/pages/index.tsx:17]:
```suggestion
  const {
    pendingUserData,
    ...functionCard
  } = usePendingUserData()
```

File: apps/web/src/modules/account/hooks/usePendingUserData/index.ts [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+    if (router.isReady) fetchPendingUserData()
+  }, [handleError, router.isReady])
+
+  const acceptUser = async () => {
+    try {
+      await apiClient.put<UpdateUserStatusDto>(
+        `/users/${modalId}/verification`,
+        {
+          status: UserStatus.ACCEPTED,
+          rejectedReason: '',
```
Comment (PKhing) [inline apps/web/src/modules/account/hooks/usePendingUserData/index.ts:40]: remove reason

File: apps/web/src/modules/account/pages/index.tsx [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+        ) : (
+          pendingUserData.map((user) => (
+            <PendingUserCard
+              key={user.data.userId}
+              setReason={setModalReason}
+              setId={setModalId}
+              accept={acceptUser}
+              reject={rejectUser}
+              data={user}
+            />
```
Comment (miello) [inline apps/web/src/modules/account/pages/index.tsx:34]:
```suggestion
            <PendingUserCard
              key={user.data.userId}
              data={user}
              {...functionCard}
            />
```

File: apps/web/src/modules/rejected/pages/index.tsx [hunk 18]
```diff
@@ -1,10 +1,12 @@
 import { UserType } from '@modela/database'
+import { ActorInfoDto, CastingInfoDto } from '@modela/dtos'
```
Comment (miello) [inline apps/web/src/modules/rejected/pages/index.tsx:2]:
```suggestion
import { ActorInfoDto, CastingInfoDto, UserType } from '@modela/dtos'
```

**Review by PKhing — CHANGES_REQUESTED** [review]

**Review by miello — CHANGES_REQUESTED** [review]

**Comment by PKhing** [comment]
ทำตัวหนา (วิธีมีเขียนไว้ใน change log แล้ว)
image

**Comment by PKhing** [comment]
bottom padding ยังไม่มีเลยครับ
image

**Comment by PKhing** [comment]
card แคบไปป่ะนะ (width ~930 px)
image

**Comment by PKhing** [comment]
top and bottom space is not balance
image
image

**Comment by PKhing** [comment]
iphone XR
image

**Review by PKhing — APPROVED** [review]

**Pull request merged**